<launch>
    <arg name="sim_time_required" default="true"/>
    <param name="use_sim_time" value="$(arg sim_time_required)"/>

    <!-- dataset config -->
    <arg name="robot_name" default="a1"/>
    <arg name="use_gt_frame" default="false"/>
    <arg name="robot_frame" default="base" if="$(arg use_gt_frame)"/>
    <arg name="robot_frame" default="base_link" unless="$(arg use_gt_frame)"/>
    <arg name="use_image_interface" default="true"/>
    <arg name="use_compressed_transport" default="true"/>

    <!-- semantics -->
    <arg name="labelspace_name" default="ade20k_full"/>
    <arg name="semantic_map_dir" default="$(find semantic_inference)/config/colors"/>
    <arg name="semantic_map_file" default="$(arg labelspace_name).csv"/>
    <arg name="semantic_map_path" default="$(arg semantic_map_dir)/$(arg semantic_map_file)"/>

    <!-- topics -->
    <arg name="rgb_info_topic" default="/cam_d455/color/camera_info"/>
    <arg name="rgb_topic" default="/cam_d455/color/image_raw"/>
    <arg name="depth_topic" default="/cam_d455/depth/image_rect_raw"/>
    <arg name="depth_info_topic" default="/cam_d455/depth/camera_info"/>
    <arg name="pointcloud_topic" value="/semantic_pointcloud"/>
    <arg name="label_topic" default="/semantic/image_raw"/>
    <arg name="prerecorded_topic" default="/oneformer/labels/image_raw"/>

    <include file="$(find hydra_ros)/launch/sources/a1_pointcloud.launch" pass_all_args="true"/>

    <include file="$(find hydra_ros)/launch/hydra.launch" pass_all_args="true">
        <arg name="config_dir" default="$(find hydra)/config/d455_outdoor"/>
        <arg name="extrinsics_type" value="kimera"/>
        <arg name="kimera_sensor_filepath" default="$(find hydra_vio_configs)/config/robots/a1_center_external/LeftCameraParams.yaml"/>
        <arg name="rviz_file" default="a1.rviz"/>
        <arg name="use_oriented_bounding_boxes" default="true"/>
        <arg name="dsg_output_prefix" default="castle_island"/>
        <arg name="image_queue_size" value="100"/>
        <arg name="rgb_image_transport" value="compressed" if="$(arg use_compressed_transport)"/>
        <arg name="rgb_image_transport" value="raw" unless="$(arg use_compressed_transport)"/>
        <arg name="depth_topic" default="depth_registered/image_rect"/>
    </include>

    <node pkg="tf2_ros" type="static_transform_publisher" name="map_odom_link" args="0 0 0 0 0 0 1 map odom" unless="$(arg use_gt_frame)"/>

</launch>
